src/main/telemetry/sbus2_sensors.c

```diff
@@ -0,0 +1,731 @@
+/*
+ * This file is part of RotorFlight.
+ *
+ * RotorFlight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * RotorFlight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with RotorFlight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "telemetry/sbus2.h"
+#include "telemetry/sbus2_sensors.h"
+
+
+#ifdef USE_TELEMETRY_SBUS2
+void send_RPM(uint8_t port, uint32_t RPM)
+{
+   uint32_t value =  0;
+   uint8_t bytes[2] = {0, 0};
+
+   value =  RPM / 6;
+   if(value > 0xffff){
+    value = 0xffff;
+   }
+   bytes[1] = value >> 8;
+   bytes[0] = value;
+   SBUS2_transmit_telemetry_data( port , bytes);
+}
+
+
+void send_temp125(uint8_t port, int16_t temp)
+{
+   int16_t value=  0;
+   uint8_t bytes[2] = {0,0};
+
+   value = temp | 0x4000;
+   bytes[0] = value >> 8;
+   bytes[1] = value;
+   SBUS2_transmit_telemetry_data( port , bytes);
+}
+void send_SBS01T(uint8_t port, int16_t temp){
+  int16_t value=  0;
+  uint8_t bytes[2] = {0,0};
+
+  value = temp | 0x8000;
+  value = value + 100;
+  bytes[0] = value;// >> 8;
+  bytes[1] = value >> 8;
+  SBUS2_transmit_telemetry_data(port , bytes);
+}
+
+void send_voltage(uint8_t port,uint16_t voltage1, uint16_t voltage2)
+{
+   uint16_t value = 0;
+   uint8_t bytes[2] = {0, 0};
+
+   // VOLTAGE1
+   value = voltage1 | 0x8000; 
+   if ( value > 0x9FFF ){
+     value = 0x9FFF; // max voltage is 819.1
+   }
+   else if(value < 0x8000){
+     value = 0x8000;
+   }
+   bytes[0] = value >> 8;
+   bytes[1] = value;
+   SBUS2_transmit_telemetry_data( port , bytes);
+
+   //VOLTAGE2
+   value = voltage2;
+   if ( value > 0x1FFF ){
+     value = 0x1FFF; // max voltage is 819.1
+   }  
+   bytes[0] = value >> 8;
+   bytes[1] = value;
+   SBUS2_transmit_telemetry_data( port+1 , bytes);
+}
+
+void send_s1678_current(uint8_t port, uint16_t current, uint16_t capacity, uint16_t voltage)
+{
+   uint16_t value = 0;
+   uint32_t local = 0;
+   uint8_t bytes[2] = {0, 0};
+
+   // CURRENT
+   local = ((uint32_t)current) * 1 ;
+   value = (uint16_t)local;   
```

Comment:
I agree. :)

Maybe it was to quiet down a compiler warning.

The * 1 does not make much sense. Maybe it was related to unit conversion at some point. I can't really see how casting to uint32_t first would change the result from just casting it to uint16_t directly,

I can simplify it if you want.

Reply:
If the file is directly from the origin, and there are no bugs... maybe just leave it for now.

Reply:
If it turn out later that it's also buggy, then I would suggest cleaning it up

Reply:
That is part of the reason I kept it separate. Easier to cross reference in the future.

There are small changes (formatting) due to copying code around and initialization changes due to compiler warnings, but otherwise, the sensor packing code is the same as the original.